genesyscloud/architect_grammar/genesyscloud_architect_grammar_proxy.go

```diff
@@ -0,0 +1,283 @@
+	if err != nil {
+		return nil, fmt.Errorf("Failed to create grammar: %s", err)
+	}
+
+	return grammar, nil
+}
+
+// createArchitectGrammarLanguageFn is an implementation function for creating a Genesys Cloud Architect Grammarlanguage
+func createArchitectGrammarLanguageFn(ctx context.Context, p *architectGrammarProxy, grammarId string, language *platformclientv2.Grammarlanguage) (*platformclientv2.Grammarlanguage, error) {
+	language, _, err := p.architectApi.PostArchitectGrammarLanguages(grammarId, *language)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to create grammar language: %s", err)
+	}
+
+	return language, nil
+}
+
+// getAllArchitectGrammarFn is the implementation for retrieving all Architect Grammars in Genesys Cloud
+func getAllArchitectGrammarFn(ctx context.Context, p *architectGrammarProxy) (*[]platformclientv2.Grammar, error) {
+	var allGrammars []platformclientv2.Grammar
+
+	for pageNum := 1; ; pageNum++ {
+		const pageSize = 100
+
+		grammars, _, err := p.architectApi.GetArchitectGrammars(pageNum, pageSize, "", "", []string{}, "", "", "", true)
+		if err != nil {
+			return nil, fmt.Errorf("Failed to get architect grammars: %v", err)
+		}
+
+		if grammars.Entities == nil || len(*grammars.Entities) == 0 {
+			break
+		}
+
+		for _, grammar := range *grammars.Entities {
+			log.Printf("Dealing with grammar id : %s", *grammar.Id)
+			allGrammars = append(allGrammars, grammar)
+		}
+	}
+
+	return &allGrammars, nil
+}
+
+// getArchitectGrammarByIdFn is an implementation of the function to get a Genesys Cloud Architect Grammar by Id
+func getArchitectGrammarByIdFn(ctx context.Context, p *architectGrammarProxy, grammarId string) (grammar *platformclientv2.Grammar, statusCode int, err error) {
+	grammar, resp, err := p.architectApi.GetArchitectGrammar(grammarId, true)
+	if err != nil {
+		return nil, resp.StatusCode, fmt.Errorf("Failed to retrieve grammar by id %s: %s", grammarId, err)
+	}
+	return grammar, resp.StatusCode, nil
+}
+
+// getArchitectGrammarIdByNameFn is an implementation of the function to get a Genesys Cloud Architect Grammar by name
+func getArchitectGrammarIdByNameFn(ctx context.Context, p *architectGrammarProxy, name string) (grammarId string, retryable bool, err error) {
+	const pageNum = 1
+	const pageSize = 100
+	grammars, _, err := p.architectApi.GetArchitectGrammars(pageNum, pageSize, "", "", []string{}, name, "", "", true)
+	if err != nil {
+		return "", false, fmt.Errorf("Error searching architect grammar %s: %s", name, err)
+	}
+
+	if grammars.Entities == nil || len(*grammars.Entities) == 0 {
+		return "", true, fmt.Errorf("No architect grammars found with name %s", name)
+	}
+
+	var grammar platformclientv2.Grammar
+	for _, grammarSdk := range *grammars.Entities {
+		if *grammarSdk.Name == name {
+			log.Printf("Retrieved the grammar id %s by name %s", *grammarSdk.Id, name)
+			grammar = grammarSdk
+			return *grammar.Id, false, nil
+		}
+	}
```

Comment:
You could reuse the `getAllArchitectGrammarFn` function here to get all the grammars. You'd just have to add name as a parameter to that function. Then for the exporter `name` can just be `""`

Reply:
👍